### ISSUE TITLE
docs(devex): local development and asset paths

### ISSUE THREAD

**Issue by mikeshultz** [ApeWorX/ape#1922]
### What went wrong?

Local docs devex is a little tricky because of how we're handling asset paths.  We use a custom template context/filter to change paths to CSS files:

https://github.com/ApeWorX/ape/blob/274986c5e390a3dc933f1145202f68014976bde2/docs/_templates/layout.html#L27-L33

The filter rewrites paths to point to the latest branch asset path on RtD:

https://github.com/ApeWorX/ape/blob/274986c5e390a3dc933f1145202f68014976bde2/docs/conf.py#L88-L98

I'm not entirely sure why this was done.  The docstring suggests to was to "save room" but I imagine every branch that build docs is still generating these files.

When writing docs locally, it's usually very useful to see how they will be displayed, with tools like `sphinx-autodoc`.  For instance:

```bash
sphinx-autobuild --watch src/ docs/ build/docs/
```

However, since the asset paths are rewritten to a RtD specific path, it means CSS files return 404s and the docs display as unformatted plain text.

### How can it be fixed?

When testing locally, I update the custom filter to look like this:

```python
def fixpath(path: str) -> str:
    """
    Change paths to reference the resources from 'latest/' to save room.
    """
    return path
```
We could either detect locally if we're running autodoc somehow, or we could stop rewriting these paths entirely and let sphinx do its thing.  Wondering if I'm missing some context however, so maybe there's a good reason we're doing this?

**Comment by fubuloubu** [comment]
Yes this originally was very custom, the basic requirements were that everything get built to one flat file structure so we can have the version selector feature work as we added each version docs to the `gh-pages` build every release (but the theme is updated every time during the build). The orignal person who built it I don't think knew sphinx all that well, so if you have better suggestions we are all ears because the docs are currently hard to maintain (something @dtdang will work on in the near future)

We were working on formalizing this build process into a sphinx plugin that we could maintain for ourselves (unless someone has already made `sphinx-ghpages`) to standardize and streamline this process
